packages/block-library/src/heading/edit.js

```diff
@@ -92,22 +90,20 @@ function HeadingEdit( {
 
 	return (
 		<>
-			{ blockEditingMode === 'default' && (
-				<BlockControls group="block">
-					<HeadingLevelDropdown
-						value={ level }
-						onChange={ ( newLevel ) =>
-							setAttributes( { level: newLevel } )
-						}
-					/>
-					<AlignmentControl
-						value={ textAlign }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { textAlign: nextAlign } );
-						} }
-					/>
-				</BlockControls>
-			) }
+			<BlockControls group="block">
```

Comment:
Changing the logic to per-control does leave an empty toolbar group:
![Screenshot 2024-01-18 at 9 34 11 am](https://github.com/WordPress/gutenberg/assets/677833/3e620230-ccc9-44f5-ae7a-ad5a54f8e9ae)

But maybe it's possible to fix that another way.